openrlhf/trainer/sft_trainer.py

```diff
@@ -221,6 +221,22 @@ def fit(self, args, consumed_samples=0, num_update_steps_per_epoch=None):
 
             epoch_bar.update()
 
+        # Save deepspeed checkpoint
+        # If we have already finished training, then we decrement the step counter that was auto incremented
+        # before the start of epoch
+        if start_epoch == self.epochs:
```

Comment:
don't we want to do this regardless? so the `if` condition should be removed.

Reply:
otherwise, the whole following section should go in the `if statement`.

Reply:
Hmm will patch